ui/src/app/api.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { AppConfig } from './app.config';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class ApiService {
+  private readonly apiUrl;
+
+  constructor(private http: HttpClient, private configuration: AppConfig) {
+    this.apiUrl = configuration.apiUrl;
+  }
+
+  getEmployees<Employee>(): Promise<Employee> {
+    return new Promise<Employee>((resolve, reject) => {
+      this.http.get<Employee>(`${this.apiUrl}/employees`).subscribe(resolve, reject);
+    });
+  }
+
+  getEmployee<Employee>(id: number): Promise<Employee> {
+    return new Promise<Employee>((resolve, reject) => {
+      this.http.get<Employee>(`${this.apiUrl}/employees/${id}`).subscribe(resolve, reject);
+    });
+  }
+
+  addEmployee<Employee>(body: object): Promise<Employee> {
+    return new Promise<Employee>((resolve, reject) => {
+      this.http.post<Employee>(`${this.apiUrl}/employees`, body).subscribe(resolve, reject);
+    });
+  }
+
+  deleteEmployee(id: number): Promise<string> {
+    return new Promise<string>((resolve, reject) => {
+      this.http.delete<string>(`${this.apiUrl}/employees/${id}`).subscribe(resolve, reject);
```

Comment:
I think observables have a toPromise method, so you don't have to wrap them with new promise. It will be better if we use rxjs over async/await as that is recommended way and observables are more powerful and can be used in angular templates directly, not very sure about if promises can work in templates.